mkosi/__init__.py

```diff
@@ -739,15 +752,61 @@ def create_image(args: CommandLineArguments, root: str, for_cache: bool) -> Opti
         disable_cow(f.name)
         f.truncate(image_size(args))
 
-        table, run_sfdisk = determine_partition_table(args)
+        exec_sfdisk(args, f)
 
-        if run_sfdisk:
-            run(["sfdisk", "--color=never", f.name], input=table.encode("utf-8"))
-            run(["sync"])
+    return f
 
-        args.ran_sfdisk = run_sfdisk
 
-    return f
+def refresh_partition_table(args: CommandLineArguments, f: BinaryIO) -> None:
+    if not args.output_format.is_disk():
+        return None
```

Comment:
Let's try to make for a more uniform style here. 
```suggestion
        return
```
Whether `None` is returned explicitly or implicitly is a bit all over the place, but the implicit returns outnumber the explicit returns 2:1, so I guess that's the way to go for new code.

Reply:
ah, i didn't know that "return" and "return None" was equivalent. Interesting.

What's the general coding style suggestions (as implemented in black and so on) about this? Is one clearly preferable to the other? What if the return type of the function is `Optional[int]`, i.e. wouldn't it make sense in such a case to do an explicit `return None`, because it's not a void function but actually one that sometimes returns something?

Reply:
All functions in Python return a value, so there isn't really the concept of a void function. Not having a return is equivalent to having a bare return, both return `None`.

What's more pythonic is, I think, harder to answer. Sure, Python generally says "Explicit is better than implicit", but grepping the stdlib, there too implicitly returning `None` outweighs explicit returns 2:1. Since type hints are still a rather young thing in the whole scheme of things Python, I don't think there's a specific culture for that yet either. I think it's mostly a thing of personal preference and at least black hasn't started regulate this yet and I only remarked on this, because there was one function with an explicit `None`-return and one with implicit ones. :)

I think implicitly returning `None` makes sense at points where the code bails out and the value of `None` is not of interest, i.e. in something close to a void function, whereas explicitly returning `None` is useful when we are interested in the actual value down the road.